File: 0074-search-a-2d-matrix/solution.c
bool searchMatrix(int** matrix, int m, int* n, int target) {
    int l=0;
    int r=m* *n-1;
    while(l<=r){
        int mid = (l+r)/2;
        int row = mid / *n ;
        int col = mid% *n;
        if(matrix[row][col]==target){
            return true;
        }else if (matrix[row][col]<target){
            l=mid+1;
        }else{
            r=mid-1;
        }
    }
    return false;
}

File: 0074-search-a-2d-matrix/solution.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int r=matrix.length;
        int c=matrix[0].length;
        int row=0;
        int col=c-1;
        while(row<r && col>=0){
            if(matrix[row][col]==target){
                return true;
            }else if (matrix[row][col]<target){
                row++;
            }else{
                col--;
            }
        } 
        return false;
    }
}
